torch/lib/THD/master_worker/master/generic/THDTensorCopy.h

```diff
@@ -4,4 +4,7 @@
 
 THD_API void THDTensor_(copy)(THDTensor *tensor, THDTensor *src);
 
+THD_API void THDTensor_(copyTH)(thpp::Tensor &from, THDTensor *to);
+THD_API void THDTensor_(copyTHD)(THDTensor *from, thpp::Tensor &to);
```

Comment:
This is all cool, but we need to maintain a C API here. You need to use a TensorDescriptor. Actually, maybe let's define only two functions `copyToWorker` and `copyFromWorker`?